[JCSnap] iP

JCSnap opened the pull request:
# BouncyBob
> I kinda like balls.

Here are some facts about BouncyBob:
- He is a chatbot
- He is bouncy, which is why he likes **balls**! 🏀

To set up, please follow these steps:
1. Download the [file](https://github.com/JCSnap/ip/releases/tag/A-Jar). Look out for the `bouncybob.jar` file!
2. Run it

Here's a checklist to try the commands:
- [ ] todo
- [ ] deadline
- [ ] event
- [ ] mark
- [ ] unmark
- [ ] delete
- [ ] list
- [ ] bye

If you are interested in how it works, here is the start of the program!
```java
public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        TaskList taskList = new TaskList();
        // rest of the code
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

JCSnap committed:
changed Duke to BouncyBob with a cooler greeting

JCSnap committed:
added feature where chatbot repeats after user and terminates after bye

JCSnap committed:
added feature to store user input and display them when asked

JCSnap committed:
added Task class

JCSnap committed:
added getters to Task

JCSnap committed:
refractored chatbot to use object Task instead of string

JCSnap committed:
added setter for undone

JCSnap committed:
added feature to set a task as done or undone

JCSnap committed:
added Deadlines class

JCSnap committed:
added Events class

JCSnap committed:
added ToDos class

JCSnap committed:
added new functions to parent class

JCSnap committed:
fixed no return error

JCSnap committed:
refractored command blocks into individual functions

JCSnap committed:
added logic for deadlines and events

JCSnap committed:
removed unused getters

JCSnap committed:
added print task count in list

JCSnap committed:
removed unecessary spaces

JCSnap committed:
added test cases and expected output

JCSnap committed:
changed filename

JCSnap committed:
handled common errors

JCSnap committed:
added error handling for mark and unmark

JCSnap committed:
refractored to use arrayList and added delete

JCSnap committed:
refractored to use enum

JCSnap committed:
Add writing tasks to hard disk

JCSnap committed:
Add loading tasks from hard disk

JCSnap committed:
Remove prints

JCSnap committed:
Add localDateTime recognition to the classes

JCSnap committed:
Add exception handling for wrong datetime format

JCSnap committed:
Merge branch 'branch-Level-8'
merged

JCSnap committed:
Refractor to use Ui class

JCSnap committed:
Refractor code to use Parser class

JCSnap committed:
Refractor code to use TaskList

JCSnap committed:
Merge pull request #1 from JCSnap/A-MoreOOP

A more oop

JCSnap committed:
Refractor into packages

JCSnap committed:
Merge pull request #2 from JCSnap/A-Package

Refractor into packages

JCSnap committed:
Merge pull request #3 from JCSnap/add-gradle-support

Add gradle support

JCSnap committed:
Merge branch 'master' of https://github.com/JCSnap/ip into branch-Level-8
merged

JCSnap committed:
Add configuration for gradle and for gradle tests

JCSnap committed:
Merge pull request #4 from JCSnap/A-Gradle

Add gradle configuration and gradle tests

JCSnap committed:
Add test cases for Deadlines class

JCSnap committed:
Merge tag 'A-JUnit' into branch-Level-8
merged

JCSnap committed:
Merge branch 'branch-Level-8'
merged

JCSnap committed:
Add javadoc for all files

JCSnap committed:
Update code to fulfill coding standards

JCSnap committed:
Add find command

JCSnap committed:
Merge branch 'branch-A-JavaDoc'
merged javadoc

JCSnap committed:
Merge branch 'branch-A-CodingStandard'
merged CodingStandard

JCSnap committed:
Merge conflict

shuenj commented on src/main/java/bouncybob/task/ToDos.java:
```diff
@@ -0,0 +1,26 @@
+/**
+ * Represents a To-Do task in the BouncyBob application.
+ */
+public class ToDos extends Task {
```
Perhaps ToDo instead of ToDos since it represents a single task? Same for the other task classes.

shuenj reviewed (commented):
LGTM, just minor naming suggestions.

awhb commented on src/main/java/bouncybob/task/Deadlines.java:
```diff
@@ -0,0 +1,58 @@
+/**
+ * Represents a deadline task in the BouncyBob application.
+ */
+public class Deadlines extends Task {
```
The Java coding standard encourages singular class names e.g. Deadline
```suggestion
public class Deadline extends Task {
```

awhb replied:
I'd encourage you to avoid plural class names as part of the Java coding standard for this course! 
```suggestion
public class Deadline extends Task {
```

awhb commented on src/main/java/bouncybob/util/TaskFileHandler.java:
```diff
@@ -0,0 +1,91 @@
+ * Utility class for handling file operations related to tasks.
+ */
+public class TaskFileHandler {
+    private static final String FILE_PATH = "./data/bouncy.txt";
```
Great work avoiding a common violation of not naming static final variables to the style for constants in the Java Coding Standard!

awhb reviewed (commented):
Test

awhb commented on src/test/java/bouncybob/util/ParserTest.java:
```diff
@@ -0,0 +1,51 @@
+
+public class ParserTest {
+
+    @Test
+    public void testGetTaskType() {
```
Do refer to the Java coding standard for Java test method naming conventions e.g. the test method sortList() will test the sortList() method for all scenarios.

awhb commented on src/main/java/bouncybob/util/TaskFileHandler.java:
```diff
@@ -0,0 +1,91 @@
+package bouncybob.util;
+
+import bouncybob.task.Deadlines;
+import bouncybob.task.Events;
+import bouncybob.task.Task;
+import bouncybob.task.ToDos;
+
+import java.io.BufferedWriter;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.IOException;
```
Your imports are consistently alphabetically ordered, avoiding a common lazy Java coding standard violation! Much appreciated!

awhb commented on src/main/java/bouncybob/util/Ui.java:
```diff
@@ -0,0 +1,139 @@
+ * Handles the user interface of the BouncyBob application.
+ */
+public class Ui {
+    private static final String TOP_BORDER = "================================================";
+    private static final String MIDDLE_BORDER = "|                                              |";
+    private static final String BOTTOM_BORDER = "================================================";
```
Great work avoiding a common Java coding standard violation of not following the naming convention for Java constants! Much appreciated!

awhb reviewed (commented):
Great work! 👍 
Minor changes requested.

JCSnap committed:
Add dependencies for javafx

JCSnap committed:
Add toggle status function

JCSnap committed:
Add GUI launcher

JCSnap committed:
Add GUI for create, delete, toggle status and view

JCSnap committed:
refractor and edits is now saved to disk

JCSnap committed:
create overload method for save to disk

JCSnap committed:
add gui package

JCSnap committed:
Refractor to arrange imports according to standard

JCSnap committed:
Merge branch 'branch-Level-10'
merged

JCSnap committed:
Add comments for functions

JCSnap committed:
Add Level-9 branch

JCSnap committed:
Merge branch 'branch-Level-9'
merged

JCSnap committed:
Add assertions to document important assumptions

Apart from relying on throwing errors, we also want to ensure that our
assumptions hold at every step of the programme.

To enable, enter java -ea BouncyBob

JCSnap committed:
Refractor by separating logic with newlines
Previously, some parts of the file are a bit unreadable, especially if
there are huge chunk of logic. This commit adds newlines to separate
different logic. Furthermore, this commit also add comments for the
start function.

JCSnap committed:
Refractor by changing class names to singular
Previously, Deadline, Event and Todo are plural, which is inconsistent
with Task. They are now changed to singular. Other files that include
them as dependencies are also changed.

JCSnap committed:
Refractor to make Deadline singular

JCSnap committed:
Refractor to make ambiguous namings more informative
Previously, the array of string gotten after user input is split is
called "parts", which does not give much information. It has been
renamed to userInputSplits.

JCSnap committed:
Merge pull request #5 from JCSnap/branch-A-Assertions

Add assertions to document important assumptions

JCSnap committed:
Resolve merge conflict

JCSnap committed:
Merge pull request #6 from JCSnap/branch-A-CodeQuality

Branch a code quality

JCSnap committed:
Remove conflict

JCSnap committed:
Merge branch 'branch-A-CodeQuality'
merged

JCSnap committed:
Add functionality to add note with command note index noteString

JCSnap committed:
Add note to descriptions

JCSnap committed:
Add method to parse note command

JCSnap committed:
Add functionality to load notes

JCSnap committed:
Merge branch 'branch-BCD-Extension'
merged note functionality

JCSnap committed:
Add introduction and table of contents

JCSnap committed:
Add quick start

JCSnap committed:
Add features

JCSnap committed:
Fix table of content

JCSnap committed:
Refractor to use stringbuilder for note

JCSnap committed:
Refractor to break down long functions

JCSnap committed:
Merge pull request #8 from JCSnap/branch-A-UserGuide

Branch a user guide

JCSnap committed:
Merge branch 'master' of https://github.com/JCSnap/ip into branch-A-Improvements

merged user guide

JCSnap committed:
Add find command

JCSnap committed:
Tidy up code quality

JCSnap committed:
Add ui

JCSnap committed:
Change Ui png name

JCSnap committed:
Finish up User Guide

JCSnap committed:
Merge pull request #9 from JCSnap/branch-A-Improvements

Branch a improvements

JCSnap committed:
Merge pull request #10 from JCSnap/branch-A-UserGuide

Branch a user guide
